src/Middleware/OutputCaching/test/MemoryOutputCacheStoreTests.cs

```diff
@@ -71,7 +71,15 @@ public async Task EvictByTag_SingleTag_SingleEntry()
         await store.EvictByTagAsync("tag1", default);
         var result = await store.GetAsync(key, default);
 
+        HashSet<string> tag1s;
+
+        // Wait for the hashset to be removed as it happens on a separate thread
+        var timeout = Task.Delay(1000);
```

Comment:
1 second is too short for CI machines. If we want to go this approach it should be longer and not a tight loop.
```csharp
var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
while (... && !cts.IsCancellationRequested)
{
    await Task.Yield();
}
cts.Cancel();
```

Reply:
Alternatively, these tests can use a custom `IMemoryCache` implementation so we can know when the eviction callback completes.